### PR TITLE
Expose rest client method for generating login tokens through MSC3882

### PR DESCRIPTION
**Pull request by stefanceriu**

### CODE REVIEW

File: MatrixSDK/MXRestClient.m [hunk 1]
```diff
@@ -906,6 +906,30 @@ - (NSString*)loginFallback;
     return loginFallback;
 }
 
+- (MXHTTPOperation*)generateLoginTokenWithSuccess:(void (^)(MXLoginToken *))success
+                                          failure:(void (^)(NSError *error))failure
+{
+    MXWeakify(self);
+    return [httpClient requestWithMethod:@"POST"
+                                    path:[NSString stringWithFormat:@"%@/login/token", apiPathPrefix]
```
Comment (ismailgulek) [inline MatrixSDK/MXRestClient.m:914]: The path is stable?

**Review by ismailgulek — APPROVED** [review]
LGTM with a note.

File: MatrixSDK/MXRestClient.m (hunk 1 above)
Comment (stefanceriu) [reply MatrixSDK/MXRestClient.m:914]: Ah, no it's not 😞 Good catch 👍

**Comment by stefanceriu** [comment]
This should be correct now although our current test synapse instance doesn't support it

**Review by ismailgulek — APPROVED** [review]

**Pull request merged**